### ISSUE TITLE
bad source address when run io_uring-udp with " -v" option

### ISSUE THREAD

**Issue by mczka** [axboe/liburing#814]
https://github.com/axboe/liburing/blob/0dd82f0c63d468c5b3d9510af642968ce6e8ee21/examples/io_uring-udp.c#L278
line 278 should be
```
		name = inet_ntop(ctx->af, &(addr->sin_addr), buff, sizeof(buff));
```
after change it's working as expected
```
$ ./io_uring-udp -p 5514 -v
received 1200 bytes 16 from 127.0.0.1:56848
received 1200 bytes 16 from 127.0.0.1:40671
received 1200 bytes 16 from 127.0.0.1:56848
received 1200 bytes 16 from 127.0.0.1:56848
received 1200 bytes 16 from 127.0.0.1:40671
received 1200 bytes 16 from 127.0.0.1:56848
```

**Comment by ammarfaizi2** [comment]
The fix looks good to me. Do you want to send a pull request?

**Comment by illiliti** [comment]
ipv6 address also needs fixing

**Comment by ammarfaizi2** [comment]
@illiliti good catch; the fix is below. Jens, can you take it directly?

I have tested it.

```patch
From a0a189fa55ccba335b24b12b47241172af9faad0 Mon Sep 17 00:00:00 2001
From: Ammar Faizi <ammarfaizi2@gnuweeb.org>
Date: Wed, 8 Mar 2023 04:11:29 +0700
Subject: [PATCH] io_uring-udp: Fix the wrong IPv6 binary to string conversion

Another io_uring-udp fix.

The verbose output shows the wrong address when using IPv6. AF_INET6
should use 'struct sockaddr_in6', not 'struct sockaddr_in'. Fix that.

Thanks to @illiliti.

Link: https://github.com/axboe/liburing/issues/814#issuecomment-1458862489
Fixes: https://github.com/axboe/liburing/issues/814
Fixes: 61d472b51e761e61c ("add an example for a UDP server")
Signed-off-by: Ammar Faizi <ammarfaizi2@gnuweeb.org>
---
 examples/io_uring-udp.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/examples/io_uring-udp.c b/examples/io_uring-udp.c
index b81a5e7c47afd9c8..220c8edb6c85ed02 100644
--- a/examples/io_uring-udp.c
+++ b/examples/io_uring-udp.c
@@ -274,8 +274,14 @@ static int process_cqe_recv(struct ctx *ctx, struct io_uring_cqe *cqe,
 		char buff[INET6_ADDRSTRLEN + 1];
 		const char *name;
 		struct sockaddr_in *addr = io_uring_recvmsg_name(o);
+		const void *ip_addr;
 
-		name = inet_ntop(ctx->af, &addr->sin_addr, buff, sizeof(buff));
+		if (ctx->af == AF_INET6)
+			ip_addr = &((struct sockaddr_in6 *)addr)->sin6_addr.s6_addr;
+		else
+			ip_addr = &addr->sin_addr.s_addr;
+
+		name = inet_ntop(ctx->af, ip_addr, buff, sizeof(buff));
 		if (!name)
 			name = "<INVALID>";
 		fprintf(stderr, "received %u bytes %d from %s:%d\n",
-- 
Ammar Faizi


```
